api/src/controller/http/comment/comment.v1.controller.ts

```diff
@@ -0,0 +1,169 @@
+import { User } from '@src/core/entities/user.entity';
+import { CommentService } from '@src/core/services/comment/comment.service';
+import {
+  CreateCommentDto,
+  DeleteCommentDto,
+  GetCommentsDto,
+  UpdateCommentDto
+} from '@src/core/services/comment/types';
+import { AppErrorCode, CustomError } from '@src/error/errors';
+
+import {
+  CommentV1PathParameter,
+  CreateCommentV1Request,
+  ListCommentsV1QueryParameter,
+  UpdateCommentV1Request
+} from '@controller/http/comment/request/comment.v1.request';
+import {
+  CommentV1FilterType,
+  CommentV1Response,
+  CreateCommentV1Response,
+  DeleteCommentV1Response,
+  ListCommentsV1Response,
+  UpdateCommentV1Response
+} from '@controller/http/comment/response/comment.v1.response';
+import { methodNotAllowed } from '@controller/http/handler';
+import { Middleware } from '@controller/http/middleware';
+
+export class CommentV1Controller {
+  constructor(
+    private readonly middleware: Middleware,
+    private readonly service: CommentService
+  ) {}
+
+  public routes = (): Router => {
+    const router: Router = Router();
+    const prefix = '/v1/comments';
+
+    router
+      .route(`${prefix}/:commentId`)
+      .all(this.middleware.issuePassport)
+      .put(this.updateComment)
+      .delete(this.deleteComment)
+      .all(methodNotAllowed);
+
+    router
+      .route(`${prefix}`)
+      .get(this.listComments)
+      .all(this.middleware.issuePassport)
+      .post(this.createComment)
+      .all(methodNotAllowed);
+
+    return router;
+  };
+
+  public createComment = async (
+    req: Request<any, any, any, CreateCommentV1Request, AppIdToken>,
+    res: Response<CreateCommentV1Response>
+  ) => {
+    const dto: CreateCommentDto = {
+      requesterIdToken: res.locals.passport,
+      movieName: req.body.movieName,
+      content: req.body.content
+    };
+    const { user, comment } = await this.service.createComment(dto);
+
+    res.send({ comment: this.buildCommentResponse(user, comment) });
+  };
+
+  public listComments = async (
+    req: Request<any, any, any, ListCommentsV1QueryParameter, any>,
+    res: Response<ListCommentsV1Response>
+  ) => {
+    const dto: GetCommentsDto = {
+      nickname: req.query.nickname,
+      movieName: req.query.movieName,
+      sortBy: req.query.sortBy,
+      direction: req.query.direction,
+      pageOffset: req.query.pageOffset,
+      pageSize: req.query.pageSize
+    };
+
+    const { users, comments, pagination } = await this.service.getComments(dto);
+
+    let filter: CommentV1FilterType | undefined = {
+      nickname: req.query.nickname,
+      movieName: req.query.movieName
+    };
+
+    const isFilterEmpty = Object.values(filter).every(
+      (value) => value === undefined
+    );
+
+    filter = isFilterEmpty ? undefined : filter;
+
+    const userMapById = new Map<string, User>(
+      Object.entries(Object.fromEntries(users.map((user) => [user.id, user])))
+    );
```

Comment:
```suggestion
const useMapById = users.reduce((acc, user) => {
  acc.set(user.id, user); // maybe acc.set(String(user.id), user) ?
  return acc;
}, new Map<string, User>());
```

array reduce를 사용하면 한번의 순회로 가능한 것 같아요.

Reply:
https://github.com/MovieReviewComment/Mr.C/pull/196/commits/bf226da211c2781dc3c0f1224e63b073257a9f92 에서 반영했습니다. 좋은 제안 감사합니다! 👍